File: docs-ios/mopubInit.m
// AppDelegate.m 
#import "AppDelegate.h"
  
@import SmaatoSDKCore; // Import SmaatoSDKCore 
@import MoPub; // Import MoPub
@implementation AppDelegate
  
- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
    MPMoPubConfiguration *sdkConfig = [[MPMoPubConfiguration alloc] initWithAdUnitIdForAppInitialization:@ <AD_UNIT_ID>"];
 
    sdkConfig.globalMediationSettings = @[];
    sdkConfig.loggingLevel = MPBLogLevelInfo;
    sdkConfig.allowLegitimateInterest = YES;
    sdkConfig.additionalNetworks = @[SMAMoPubBaseAdapterConfiguration.class];
     
    NSMutableDictionary *smaatoConfig = [NSMutableDictionary new];
    [smaatoConfig setObject:@"<Publisher_Id>" forKey:@"publisherId"];
    [smaatoConfig setObject:@(YES) forKey:@"httpsOnly"];
    [smaatoConfig setObject:@(NO) forKey:@"loggingDisabled"];
    [smaatoConfig setObject:@(kSMALogLevelError) forKey:@"logLevel"];
    [smaatoConfig setObject:@(kSMAMaxAdContentRatingUndefined) forKey:@"maxAdContentRating"];
 
    sdkConfig.mediatedNetworkConfigurations = [@{@"SMAMoPubBaseAdapterConfiguration":smaatoConfig} mutableCopy];
 
    [[MoPub sharedInstance] initializeSdkWithConfiguration:sdkConfig completion:^{
        NSLog(@"SDK initialization complete");
        // SDK initialization complete. Ready to make ad requests.
        SmaatoSDK.gpsEnabled = YES;
        SmaatoSDK.userGender = kSMAGenderFemale;
        SmaatoSDK.userAge = @30;
        SmaatoSDK.userSearchQuery = @"yoga, vegan, san+francisco";
    }];
     
    return YES;
}
  
@end

File: docs-ios/mopubInit.swift
// AppDelegate.swift
import UIKit
import SmaatoSDKCore
  
@UIApplicationMain
class AppDelegate: UIResponder, UIApplicationDelegate {  
    var window: UIWindow?
      
    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        
        let sdkConfig = MPMoPubConfiguration(adUnitIdForAppInitialization: "<AD_UNIT_ID>")
        sdkConfig.globalMediationSettings = []
        sdkConfig.loggingLevel = .info
        sdkConfig.allowLegitimateInterest = true
        sdkConfig.additionalNetworks = SMAMoPubBaseAdapterConfiguration()
        
        var smaatoConfig: [AnyHashable : Any] = [:]
        smaatoConfig.setObject("Publisher_ID", forKey: "publisherId")
        smaatoConfig.setObject(true, forKey: "httpsOnly")
        smaatoConfig.setObject(false, forKey: "loggingDisabled")
        smaatoConfig.setObject(kSMALogLevelError, forKey: "logLevel")
        smaatoConfig.setObject(kSMAMaxAdContentRatingUndefined, forKey: "maxAdContentRating")
        
        sdkConfig.mediatedNetworkConfigurations = [
        "SMAMoPubBaseAdapterConfiguration": smaatoConfig
        ]
        
        MoPub.sharedInstance().initializeSdk(with: sdkConfig) {
            print("SDK initialization complete.")
            SmaatoSDK.gpsEnabled = true
            SmaatoSDK.userGender = kSMAGenderFemale
            SmaatoSDK.userAge = 30
            SmaatoSDK.userSearchQuery = "yoga, vegan, san+francisco"
        }
         
        return true
    }
}